### PR TITLE
Make using tools easy with OpenAITool and Function -> OpenAITool automatic conversion

### PR DESCRIPTION
**Pull request by willbakst**

### CODE REVIEW

**Comment by brenkao** [comment]
tools link missing in mkdocs.yml

File: mirascope/chat/tools.py [hunk 1]
```diff
@@ -24,7 +31,7 @@ def tool_schema(cls) -> ChatCompletionToolParam:
             raise ValueError("Tool must have a dosctring description.")
```
Comment (brenkao) [inline mirascope/chat/tools.py:31]: typo dosctring

File: mirascope/chat/tools.py [hunk 2]
```diff
@@ -42,3 +49,38 @@ def tool_schema(cls) -> ChatCompletionToolParam:
...
+    """
+
+    def decorator(cls: Type[T]) -> Type[T]:
+        """A decorator for adding a function to a tool class."""
+
+        def fn_property(self) -> Callable:
+            """Returns the function that the tool describes."""
+            return fn
+
+        setattr(cls, "fn", property(fn_property))
```
Comment (brenkao) [inline mirascope/chat/tools.py:83]: do you need to define fn_property. could you pass fn to setattr instead?

File: mirascope/chat/utils.py [hunk 3]
```diff
@@ -15,3 +20,63 @@ def get_openai_chat_messages(
...
+        if not docstring_param.description:
+            raise ValueError("All parameters must have a description.")
+
+        field_info = {"description": docstring.params[i].description}
+        if parameter.default != Parameter.empty:
+            field_info["default"] = parameter.default
+
+        field_definitions[parameter.name] = (
+            parameter.annotation,
+            FieldInfo(**field_info),  # type: ignore
```
Comment (brenkao) [inline mirascope/chat/utils.py:74]: this ignore does not seem necessary?

Comment (brenkao) [inline mirascope/chat/utils.py:74]: nit: can we define field info as FieldInfo passing in description and default instead of unpacking dict

**Review by brenkao — COMMENTED** [review]
tools link missing in mkdocs.yml

File: mirascope/chat/tools.py (hunk 2 above)
Comment (willbakst) [reply mirascope/chat/tools.py:83]: You need to define self, but we can further simplify using a lambda

**Pull request merged**